datafusion/src/execution/dataframe_impl.rs

```diff
@@ -488,6 +548,17 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn register_table() -> Result<()> {
+        let df = test_table().await?.select_columns(&["c1", "c3"])?;
+        let mut ctx = ExecutionContext::new();
+        let df_impl = DataFrameImpl::new(ctx.state.clone(), &df.to_logical_plan());
+        ctx.register_table("test_table", Arc::new(df_impl))?;
+        let table = ctx.table("test_table")?;
+        assert_same_plan(&table.to_logical_plan(), &table.to_logical_plan());
```

Comment:
did you mean to compare `table.to_logical_plan()` with `df.to_logical_plan()` instead?

Reply:
I did! Thanks and nice catch.